int_scan.cpp

```diff
@@ -0,0 +1,133 @@
+#include "int_scan.hpp"
+
+enum Base {OCT=8, DEC=10, HEX=16};
```

Comment:
`Base` is a very generic name and it may be used elsewhere. In this case, you can just have the numeric values instead of an `enum` as it is pretty understandable.